### ISSUE TITLE
[Feature Request]: Horizontal Stretch Graphic Setting

### ISSUE THREAD

**Issue by PsychoticFox** [PCSX2/pcsx2#8628]
### Description

Basically the same setting as vertical stretch feature, but it stretches and squashes the screen horizontally.

### Reason

Some games or widescreen hacks end up looking weird or squished, sometimes in a way that can't exactly be solved with vertical stretch or "fit to window/fullscreen", or these methods won't fill the screen properly, and having a horizontal stretch setting could be a good solution to these kinds of issues. Plus it's just weird in general that a vertical stretch setting exists but not a horizontal stretch. You'd think both with exist side by side.

### Examples

Basically this but with horizontal stretch
![bandicam 2023-04-09 15-51-12-899](https://user-images.githubusercontent.com/69675052/230798375-757cccc6-01c8-41e4-ba07-8217fd9583f4.jpg)

**Comment by jonasos3** [comment]
Seconding this Issue. The addition of horizontal stretch (+ vertical) would also allow for/replace the Zoom functionality that was in Wx before. Also would help with unwanted overscan and stuff when using certain HDMI-AV adapters for CRT-usage, offering more precise control over the final image.

**Comment by stenzek** [comment]
I'd be more inclined to remove vertical than add horizontal. The crop options fulfil most use cases, and are more flexible.

**Comment by hohohiheng** [comment]
+1 I wish this is added. This is basically the custom aspect ratio equivalent in retroarch. It's very useful when I'm using my crt tv as display, because some games are simply would not fit inside my screen eventhough the resolution is already set to native and the aspect ratio is set to 4:3/stretch to screen. The crop function is totally the opposite of what I needed because its just cut out the game's display.

**Comment by refractionpcsx2** [comment]
> +1 I wish this is added. This is basically the custom aspect ratio equivalent in retroarch. It's very useful when I'm using my crt tv as display, because some games are simply would not fit inside my screen eventhough the resolution is already set to native and the aspect ratio is set to 4:3/stretch to screen. The crop function is totally the opposite of what I needed because its just cut out the game's display.

then why not use "stretch" that's already there? If you're using a CRT and the picture is to those bounds, then stretch will make it fit your CRT.

**Comment by bigol83** [comment]
> I'd be more inclined to remove vertical than add horizontal. The crop options fulfil most use cases, and are more flexible.
> 
> That's exactly why zoom was removed - it's useless when you can just crop instead.

Sorry for quoting this old message, but cropping creates bigger black borders, a zoom option would be great to fill in those borders.

**Comment by F0bes** [comment]
Closing due to reasons mentioned above.

**Comment by TrumpetMonkey** [comment]
Hey, this would be really useful for my specific use case or anything that's like it.

I could use many combinations of settings to play God of War but, it's always a compromise (assume widescreen patch is being used in all of the below cases)

-Simply play with progressive scan and widescreen on, but screen offsets off (Cutscenes have that intrusive black bar)
![Image](https://github.com/user-attachments/assets/47eeaac4-a44f-4e06-936f-0f909d20452d)

-Screen offsets with Widescreen in-game gets rid of the black bar and gives you 16:9, but you can't use progressive scan
![Image](https://github.com/user-attachments/assets/ea875ed0-6838-40cf-8b88-677837998716)
![Image](https://github.com/user-attachments/assets/ddd29b09-8b51-4903-a4f2-ca07a8c98fea)

-Screen offsets with Progressive scan (requires these pixel crops left 40 up 15 right 41 down 17) and widescreen, but the output is no longer 16:9 and has to be horizontally stretched back. There is no way I have found of doing this right now in the emulator
![Image](https://github.com/user-attachments/assets/920e534e-9d5a-4f96-966b-3b05b9c86878)

At the moment, the only way I can achieve this is Reshade as vertical squeeze gives me black bars that don't crop in, and stretch to screen doesn't work with the cropped pixels
![Image](https://github.com/user-attachments/assets/3fca0878-8deb-459c-8fc1-1fa359b5ae2d)
This is a massive compromise as it's stretching square pixels and blurring the output.

TL;DR some solutions I propose are
-having stretch to screen work with the pixel cropped output
-introducing horizontal stretch

**Comment by arturbogdan** [comment]
Seconded, please at least let the cropping + stretch work together, as currently you can't really use a custom ratio.
